app/routes/book_routes.py

```diff
@@ -23,8 +23,30 @@ def create_book():
 
 @books_bp.get("")
 def get_all_books():
-    query = db.select(Book).order_by(Book.id)
-    books = db.session.scalars(query)
+    query = db.select(Book)
+
+    # If we have a `title` query parameter, we can add on to the query object
+    title_param = request.args.get("title")
+    if title_param:
+        # Match the title_param exactly, including capitalization
+        # query = query.where(Book.title == title_param)
+
+        # If we want to allow partial matches, we can use the % wildcard with `like()`
+        # If `title_param` contains "Great", the code below will match 
+        # both "The Great Gatsby" and "Great Expectations"
+        # query = query.where(Book.title.like(f"%{title_param}%"))
+
+        # If we want to allow searching case-insensitively, 
+        # we could use ilike instead of like
+        query = query.where(Book.title.ilike(f"%{title_param}%"))
+
+    # If we have other query parameters, we can continue adding to the query
```

Comment:
Maybe this a good spot to draw attention to fact that the previous query variable is updated to point at the result of adding the earlier filter? Some of this should probably just be in the text, but we could include verbiage like:

```suggestion
    # If we have other query parameters, we can continue adding to the query.
    # Notice that the previous filtering reassigned the query variable when adding
    # an additional clause. This gives us a reference to the query that includes
    # that clause. If we add an additional clause now for the description, the final
    # query will include both the title and description clauses. We should get in the
    # habit of updating the query variable as we add additional clauses, even when
    # it's currently the last one, so that if we add additional filters in the future,
    # we don't forget to update the query variable. Forgetting to do so can lead
    # to a frustrating debugging session to figure out why the filters are being
    # applied inconsistently!
```

Reply:
I think a short comment in the code would be really helpful, something like:
```
# If we have other query parameters, we can continue adding to the query.
# We need to reassign the query variable to capture the new clause we are adding.
# If we don't reassign query, we are calling the `where` function but are not saving the resulting filter
```

and as you mentioned, we could add the more detailed description to the lesson content.

Reply:
That sounds good to me. The only suggestion I'd make is that since this comment comes after where the reassignment occurs, modify that second sentence to call back to what we did.
```
# If we have other query parameters, we can continue adding to the query.
# As we did above, we must reassign the query variable to capture the new clause we are adding.
# If we don't reassign query, we are calling the `where` function but are not saving the resulting filter
```